nav2_graceful_controller/include/nav2_graceful_controller/parameter_handler.hpp

```diff
@@ -44,12 +45,13 @@ struct Parameters
   double v_linear_max_initial;
   double v_angular_max;
   double v_angular_max_initial;
+  double v_angular_min_in_place;
   double slowdown_radius;
-  bool initial_rotation;
-  double initial_rotation_min_angle;
-  bool final_rotation;
+  double initial_rotation_tolerance;
+  bool prefer_final_rotation;
   double rotation_scaling_factor;
   bool allow_backward;
+  bool add_orientations;
```

Comment:
We need orientations for any pose that ends up being a candidate for the target_pose of the control law (in order to simulate the control law - even if we end up discarding that candidate and moving to another one when the simulation fails) - so we need orientations on far more than just the back of the path.

As for "check if non-zero" - since there are default values for geometry_msgs::msg::Quaternion, we no longer end up with invalid quaternions in ROS 2. Therefore, I'm not sure we want to magically change any orientation that is basically a "zero" yaw angle. For instance, something like the lattice planner uses a set of discrete headings - and so a fairly large number of poses in the path will have "zero" headings, and we probably don't want to re-orient those towards the next pose in the path since that may no longer actually be a valid path if your robot is non-circular.

Reply:
What I'm thinking of is to check first if all path points except the start and end `i = 1 -> N-1` and if **all** are `0`, then I think its a safe bet that its not kinematically feasible. I don't think we should change individual ones, but all-or-nothing based on that.